src/asynch/state.rs

```diff
@@ -28,48 +29,56 @@ pub enum PowerState {
     PowerUp,
     Alive,
     Initialized,
+    Connected,
```

Comment:
What is the reasoning behind having `Connected` and `DataEstablished` as port of an enum called `PowerState`?

Reply:
We could set it as a constant and if it is ever not enough, just increase it would be a lot less generic parameters pain. But if we want then to be able to change it and keep the API, we would then need a feature set for that or something like that.


Ah, wanted to rename that to `OperationSate` as it could never be connected but powered off so having these in one enum would make sense.

Reply:
Makes sense :+1: